woocommerce/woocommerce-admin
# Decouple date package from `wcSettings`.

Fixes #3030.

This PR seeks to remove the `wcSettings` dependency from the Date package.

### `Date`

The exported methods of the date package that depended on `getSetting()` have been rewritten to accept a configuration object as their second parameter - this allows for composition using lodash's `partialRight()` - which is done in the new `lib/date` module.

`loadLocaleData()` is also no longer called within the date package, but in the `lib/date` module instead.

A new `lib/date` module has been introduced that exports the same API as the previous version of `@woocommerce/date` so that the consuming parts of the WC-Admin app don't need to change (too much).

### `AdvancedFilters` and `ReportFilters`

Add a new required `storeDate` prop as a means to pass down date initialization values from `@woocommerce/date`, but from `lib/date` in the app.

### Detailed test instructions:

- Verify that date values are properly formatted on all WC-Admin screens (check activity panels, reports, filters, etc)
- Verify that the default date range analytics setting functions
- Review changes to documentation



### Changelog Note:



Fix: decouple Date package from global wcSettings object.
